apps/test/unit/templates/curriculumCatalog/CurriculumCatalogTest.jsx

```diff
@@ -121,4 +68,175 @@ describe('CurriculumCatalog', () => {
       }
     });
   });
+
+  it('filtering by grade level shows any shown course that supports one of the selected grades', () => {
+    const numTotalCurriculumCards = screen.getAllByText('Quick View', {
+      exact: false,
+    }).length;
+    expect(numTotalCurriculumCards).to.equal(allShownCurricula.length);
+
+    // Select "Kindergarten" and "Grade 2" in grade level filter
+    const kindergartenFilterCheckbox = screen.getByDisplayValue('kindergarten');
+    fireEvent.click(kindergartenFilterCheckbox);
+    assert(kindergartenFilterCheckbox.checked);
+    const grade2FilterCheckbox = screen.getByDisplayValue('grade_2');
+    fireEvent.click(grade2FilterCheckbox);
+    assert(grade2FilterCheckbox.checked);
+
+    // Filters for all courses for kindergarten and/or grade 2
+    const numFilteredCurriculumCards = screen.getAllByText('Quick View', {
+      exact: false,
+    }).length;
+    expect(numFilteredCurriculumCards).to.equal(
+      gradesKAnd2ShownCurricula.length
```

Comment:
verifying the length seems like a good start, but also seems a little too easy for a bug to slip through the cracks. what I would suggest is something like:
1. add a human-readable classname to the h4 tag containing the title of the course in the DOM:
![Screenshot 2023-05-11 at 11 04 11 AM](https://github.com/code-dot-org/code-dot-org/assets/8001765/b0dd0185-8a1d-4b2f-81ca-738e56c9b206)
2. map the results of `getAllByText` and `gradesKAnd2ShownCurricula` to both contain the list of courses
3. sort them and expect the two lists of course names to be identical

Reply:
Oooo yes that sounds great, I'll do that!